Fix broken ADC test for arduino_101_sss and quark_d2000

dcpleung opened the pull request:
This fixes the adc_api test failures (#12632) on both arduino_101_sss and quark_d2000. Also, since the `adc_dw` driver is tied to the Quark SE C1000 SoC, it cannot be used for a vanilla DesignWare IP block anymore. So rename the driver too.

anangl commented on drivers/adc/adc_intel_quark_d2000.c:
```diff
@@ -238,8 +238,8 @@ static int adc_quark_d2000_read_request(struct device *dev,
 	if (seq_tbl->options &&
-	    seq_tbl->options->extra_samplings &&
-	    seq_tbl->options->interval_us < 10) {
+	    (seq_tbl->options->interval_us > 0) &&
```
When there is only one sampling to be done, there is no point in checking the interval as it won't be used anyway. I guess that's why `extra_samplings` was checked here. And I guess this check could stay, only this proper handling of zero interval should be added (I mean `(seq_tbl->options->interval_us > 0)`).

dcpleung replied:
The whole if statement is incorrect. It was "kind of" inherited from previous adc_dw implementation before the ADC change. So these are no longer valid.

dcpleung committed:
adc: overhaul adc_dw and rename it to adc_intel_quark_se_c1000_ss

Since commit 0906a51dacc947c5fa19a80e208549f9039556cd, the driver
fails the test: tests/drivers/adc/adc_api. Some of the workflow
turns out to be incorrect (e.g. not doing dummy conversion after
getting out of deep power down, and not clearing interrupt status
bits). So take some time to overhaul the driver. Also rename
the driver to adc_intel_quark_se_c1000_ss because the inner
working of this driver is tied to Quark SE C1000 SoC.

Fixes: #12632

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
adc: quark_d2000: fix sampling conversion

The logic in sampling is incorrect. The code sets up the hardware
to do multiple conversions, but in reality, each call is simply
one conversion. So fix it.

Fixes #12632

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
